### ISSUE TITLE
Avoid prefer-destructuring for unknown indexes

### ISSUE THREAD

**Issue by Zenoo** [eslint/eslint#14678]
**What rule do you want to change?**

prefer-destructuring

**Does this change cause the rule to produce more or fewer warnings?**

Fewer warnings

**How will the change be implemented? (New option, new default behavior, etc.)?**

A new option or a default behavior would both work

**Please provide some example code that this change will affect:**

```js
const array = [ { a: 1, b: 1, c: 1}, { a: 2, b: 2, c: 2}, { a: 3, b: 3, c: 3} ];
const item = array[unknownIndex]; // Here, eslint tells us to use destructuring, I don't think it should

```

**What does the rule currently do for this code?**

The current rule wants us to use destructuring when accessing an array's unknown index.

**What will the rule do after it's changed?**

The rule shouldn't ask us to use destructuring

**Are you willing to submit a pull request to implement this change?**

Yes, if I get some guidance on how-to ^^

Here's a link to a previous discussion about this:
https://github.com/eslint/eslint/discussions/14669

Here's an actual example of the rule:

![](https://camo.githubusercontent.com/6206c4cf2e3d743a15c475991b436d8c6a58e2c6fba1c8f402fb8834d74edbdd/68747470733a2f2f692e696d6775722e636f6d2f7a317a696b6c4b2e706e67)

**Comment by mdjermanovic** [comment]
Hi @Zenoo, thanks for the issue!

In the provided example, the rule suggests the use of _object_ destructuring:

```js
const { [unknownIndex]: item } = array;
```

The option to disable this warning already exists, you can set `"enforceForRenamedProperties": false`.

(it's `false` by default, you probably have `"enforceForRenamedProperties": true` in your configuration)

**Comment by Zenoo** [comment]
Oh wow, I had no idea you could destructure an unknown index with this syntax !  
Thanks a lot, problem solved !
